### PR TITLE
Makes maintenance drones no longer emaggable

### PR DESCRIPTION
**Pull request by Jaraci**
See title. I think I can count on one hand the number of times a maintenance drone has been 'legitimately' emagged, rather than just running up to the nearest antagonist and wiggling back and forth at them so they can play as an antag for free.

### CODE REVIEW

**Comment by Jaraci** [comment]
!review

File: code/modules/mob/living/silicon/robot/drone/drone.dm
Comment (SleepyGemmy) [inline code/modules/mob/living/silicon/robot/drone/drone.dm:298]: is this used for drones in the drone takeover random event?

**Review by SleepyGemmy — APPROVED** [review]

**Review by Ben10083 — COMMENTED** [review]
I really feel like this is solving a minor problem with a sledgehammer.

- Emagged drones are not happening often enough where determining if they should be removed can really be seen
- Drones are very fragile, and as such are often a straight downgrade compared to emagging a cyborg
- If you see drones trying to get themselves emagged, ahelp.

**Comment by Jaraci** [comment]
The maintainers can discuss the PR all they want, I don't have any issue with that. It's not on me whether they close it or not, but I want to address some stuff here.

Emagged drones might be fragile, but they are extremely fast (or is that just the matriarch?) compared to regular human movespeed, they have all-access, and their sprite layers underneath a lot of floor effects like catwalks and grates to make them very hard to click even if you do catch up to them. They also have a jetpack so they can operate in EVA with impunity.

It is almost impossible to prove that a maint drone is getting itself emagged deliberately, but at the same time I would contend that there's basically no way they _would_ ever get emagged without hanging around obvious antags essentially baiting it. It's an open secret that drone enjoyers love it when they get given a green light to antag; it's just borderline impossible for staff to judge without having observed the drone player for a decent time prior to them actually getting swiped.

**Comment by Ben10083** [comment]
> Emagged drones might be fragile, but they are extremely fast (or is that just the matriarch?) compared to regular human movespeed, they have all-access, and their sprite layers underneath a lot of floor effects like catwalks and grates to make them very hard to click even if you do catch up to them. They also have a jetpack so they can operate in EVA with impunity.

Just wanted to talk about this to clear the confusion.

The fastness is a matrix upgrade the matriarch can select for all drones to have, otherwise the average matriarch/maint drone is about the same speed as a human, perhaps a bit slower.

As for the sprite layer thing, this can be the problem, and it may be best to explore solutions in terms of how to click 'hidden' objects more easily, with also taking into consideration that hiding under the grates should still have a tangible effect. 

In regards to the jetpack, it has a limited amount of 'air' and to my knowledge it does not replenish (and perhaps it should, will need to look into it).

**Review by JohnWildkins — APPROVED** [review]
code fine but would prefer personally if you could still emag the fabricator

**Comment by NonQueueingMatt** [comment]
Unwanted.

**Pull request closed without merging**
